msticpy/data/queries/kql_sent_linuxsyslog_activity.yaml

```diff
@@ -0,0 +1,117 @@
+metadata:
+  version: 1
+  description: Linux Syslog Host Activity Queries
+  data_environments: [LogAnalytics]
+  data_families: [LinuxSyslog]
+  tags: ['linux', 'syslog']
+defaults:
+  metadata:
+    data_source: 'linux_syslog'
+  parameters:
+      table:
+        description: Table name
+        type: str
+        default: 'Syslog'
+      start:
+        description: Query start time
+        type: datetime
+      end:
+        description: Query end time
+        type: datetime
+      add_query_items:
+        description: Additional query clauses
+        type: str
+        default: ''
+      subscription_filter:
+        description: Optional subscription/tenant filter expression
+        type: str
+        default: 'true'
+      query_project:
+        description: Column project statement
+        type: str
+        default: '
+          | project
+            TenantId,
+            Computer,
+            Facility,
+            TimeGenerated,
+            HostName,
+            SeverityLevel,
+            SyslogMessage,
+            ProcessID,
+            ProcessName,
+            HostIP,
+          | extend TimeCreatedUtc=TimeGenerated'
+sources:
+  sudo_activity:
+    description: All sudo activity
+    args:
+      query: '
+        {table}
+        | where {subscription_filter}
+        | where TimeGenerated >= datetime({start})
+        | where TimeGenerated <= datetime({end})
+        | where Computer == "{host_name}"
+        | where ProcessName == "sudo"
+        | extend SudoResult = iif(SyslogMessage contains "authentication failure", "Failure", "Sucess")
+        | extend Sudoer = extract("(by |user=)([[:alnum:]]+)",2,SyslogMessage), SudoTo = extract("for user ([A-Z,a-z,0-9]+)",1,SyslogMessage),Command=extract("COMMAND=(.*)$",1,SyslogMessage), CommandCall=extract("COMMAND=([[:graph:]]*)",1,SyslogMessage)
+        | where SyslogMessage contains "{user}" or Sudoer contains "{user}" or SyslogMessage contains "session closed for user root"'
```

Comment:
I think I used account_name in some of the other queries, rather than "user". It would be nice if we could accept aliases for these kinds of things - maybe we can do this in the parameter extractor logic. I think I do something analogous with catching datetime data types and treating them differently. Not suggesting that you should do this work now btw - just musing over what to do.

Reply:
changed this to account_name for consistency